packages/dbgpt-core/src/dbgpt/agent/util/react_parser.py

```
@@ -197,7 +197,33 @@ def parse(self, text: str) -> List[ReActStep]:
         if not thought_matches:
-            return []
+            # Fallback: if no Thought: prefix found, try splitting on
+            # Action: prefix so we can still parse responses that omit
+            # Thought:.  Each step starts from the text before Action: (to
+            # capture Action Intention / Action Reason) up to the next
+            # Action: or end of text.
+            action_matches = self._find_prefix_matches(text, self.action_prefix_escaped)
+            if not action_matches:
+                return []
+            steps = []
+            for i, match in enumerate(action_matches):
```

Ruff (0.15.21)

[warning] Loop control variable `match` not used within loop body

(B007)